packages/runtime-core/src/renderer.ts

```diff
@@ -1139,9 +1139,7 @@ function baseCreateRenderer(
           parentSuspense,
           isSVG
         )
-        if (__DEV__ && parentComponent && parentComponent.type.__hmrId) {
-          traverseStaticChildren(n1, n2)
-        }
+        traverseStaticChildren(n1, n2)
```

Comment:
Bug caused here. 
Should always inherit `el` form previous nodes, the logic ignored `comment` case inside `traverseStaticChildren `.
I think  the hoisted `comment` is also can fix this and also refeactor the implement of codegen of `comment`.